compliance/collection/nodeinventorizer/nodeinventory.go

```diff
@@ -13,30 +16,39 @@ import (
 	"golang.org/x/exp/maps"
 )
 
+type nodeAnalyzer func(nodeName string, rootFSdir string, opts nodes.AnalyzeOpts) (*nodes.Components, error)
+
 // NodeInventorizer is the interface that defines the interface a scanner must implement
 type NodeInventorizer interface {
 	Scan(nodeName string) (*storage.NodeInventory, error)
 }
 
 // NodeInventoryCollector is an implementation of NodeInventorizer
 type NodeInventoryCollector struct {
+	analyzerFunction     nodeAnalyzer
+	analyzerCallDeadline time.Duration
+}
+
+// NewNodeInventoryCollector returns a ready to use instance of NodeInventoryCollector
+func NewNodeInventoryCollector() *NodeInventoryCollector {
+	return &NodeInventoryCollector{
+		analyzerFunction:     nodes.Analyze,
+		analyzerCallDeadline: 10 * time.Second,
```

Comment:
Have we considered making this configurable by an environment variable?

One potential reason is node scanning taking a little longer than we expected for reasons external to our control, but customers want the option to wait a little longer instead of failing.

Reply:
Covered in [31b82a7](https://github.com/stackrox/stackrox/pull/5148/commits/31b82a728cbb1134c9aa0305070fecaed2180812)